### PR TITLE
[MIG] 9.0 hr_period

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
HR Payroll Period
=================

Adds the concept of the Payroll period.

The objective is not to restrict or complicate the payroll workflow.
It is meant to reduce the number of clicks related to date selections and avoid mistakes.

Adds the date of payment on the payslip and payslip batch. This date is automatically filled when selecting a period.

And also adds a sequence on the payslip batch name and also adds the company on the payslip batch.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why don't you use date_range module as base?

**Comment by serpentcs-dev1** [comment]
IMHO this is a better way to have a new model fiscal year which gives more flexibility to generate the payroll period and to define its date of payment for that particular year.

File: hr_period/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from . import models
```
Comment (AaronHForgeFlow) [inline hr_period/__init__.py]: Please use the short license format

**Review by AaronHForgeFlow — COMMENTED** [review]
I tested on runbot and it works fine. Just the minor comment. 

BTW @pedrobaeza what is the data_range module? I searched but did not find any. Thanks in advance

**Comment by pedrobaeza** [comment]
date_range is a generic module for defining interval of dates: https://github.com/OCA/server-tools/tree/9.0/date_range

**Comment by AaronHForgeFlow** [comment]
I agree with @pedrobaeza, I checked the data range module and it fits perfectly :cool:

**Comment by AaronHForgeFlow** [comment]
@darshan-serpent I'll propose a PR to your branch

**Comment by AaronHForgeFlow** [comment]
Superseded by https://github.com/OCA/hr/pull/354

**Pull request closed without merging**